lib/txgh/github_api.rb

```diff
@@ -33,6 +33,24 @@ def create_ref(repo, branch, sha)
       client.create_ref(repo, branch, sha) rescue false
     end
 
+    def update_contents(repo, branch, content_map, message)
+      content_map.each do |path, new_contents|
+        branch = Utils.relative_branch(branch)
+        file = client.contents(repo, { path: path, ref: branch})
+        current_contents = Base64.decode64(file[:content])
+        current_sha = file[:sha]
+
+        new_sha = Utils.git_hash_blob(new_contents)
+        options = { branch: branch }
+
+        if current_sha != new_sha
```

Comment:
Cool. +1

On Wed, Jun 29, 2016, 8:55 AM Cameron Dutro <notifications@github.com>
wrote:

> In lib/txgh/github_api.rb
> <https://github.com/lumoslabs/txgh/pull/42#discussion_r68974295>:
>
> > @@ -33,6 +33,24 @@ def create_ref(repo, branch, sha)
> >        client.create_ref(repo, branch, sha) rescue false
> >      end
> >
> > +    def update_contents(repo, branch, content_map, message)
> > +      content_map.each do |path, new_contents|
> > +        branch = Utils.relative_branch(branch)
> > +        file = client.contents(repo, { path: path, ref: branch})
> > +        current_contents = Base64.decode64(file[:content])
> > +        current_sha = file[:sha]
> > +
> > +        new_sha = Utils.git_hash_blob(new_contents)
> > +        options = { branch: branch }
> > +
> > +        if current_sha != new_sha
>
> Yeah, it would. The thing is, we need to get the original blob SHA anyway
> because the update_contents API requires it (functions as a locking
> mechanism). I figured we might as well check the SHA since we need it
> anyway.
>
> —
> You are receiving this because you are on a team that was mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/lumoslabs/txgh/pull/42/files/cd183d7d18f8c48ec9a4fa3e4fd1f79744363b45#r68974295>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AAVQ3_jhTK6a5SoqZ-NFtJzFOU7pJoPrks5qQpWAgaJpZM4JAlaM>
> .
>